### ISSUE TITLE
[BUG] Cannot send mail via relayhost

### ISSUE THREAD

**Issue by ShinoYasx** [docker-mailserver/docker-mailserver#1973]
# Bug Report

## Context

When I try to send a mail using a relay smtp server (I'm using Mailjet, but I tried with Gmail), I got an SASL authentication failure: No worthy mechs found:

```
May 16 18:55:13 mail postfix/smtp[2648]: warning: SASL authentication failure: No worthy mechs found
May 16 18:55:13 mail postfix/smtp[2648]: B34B387A85506: to=<user@domain.tld>, relay=in-v3.mailjet.com[104.199.96.85]:587, delay=0.57, delays=0.07/0.06/0.45/0, dsn=4.7.0, status=deferred (SASL authentication failed; cannot authenticate to server in-v3.mailjet.com[104.199.96.85]: no mechanism available)
```
Where `user@domain.tld`is the recipient email.

I'm running docker-mailserver under armv7, and is seems that it is working fine when I'm using x86 host.

### What is affected by this bug?

I can't send mail at all because my ISP is blocking port 25, so I need to use mail relay.

### When does this occur?

It occur every time I try to send a mail using a relay host using the armv7 docker image.

### How do we replicate the issue?

1. Pull an armv7 image and use qemu if you're not running on a armv7 host
2. Fill environment variables with relay host parameters
3. Add a user account
3. Create a container with docker-compose
4. Send a mail with the registered user account

### Actual Behavior

Defer the email transmission throwing the following error:

```
May 16 18:55:13 mail postfix/smtp[2648]: Anonymous TLS connection established to in-v3.mailjet.com[104.199.96.85]:587: TLSv1.2 with cipher ADH-AES256-GCM-SHA384 (256/256 bits)
May 16 18:55:13 mail postfix/smtp[2648]: warning: SASL authentication failure: No worthy mechs found
May 16 18:55:13 mail postfix/smtp[2648]: B34B387A85506: to=<user@domain.tld>, relay=in-v3.mailjet.com[104.199.96.85]:587, delay=0.57, delays=0.07/0.06/0.45/0, dsn=4.7.0, status=deferred (SASL authentication failed; cannot authenticate to server in-v3.mailjet.com[104.199.96.85]: no mechanism available)
```

### Expected Behavior

Should send the email through the mail relay without any issue.

## Your Environment

- version: `9.1.0`
- available RAM: `2GB`
- Docker version: `20.10.6`
- docker-compose version: `1.21.0`
- host architecture: `armv7`

### Environment Variables

```CFG
DEFAULT_RELAY_HOST=[in-v3.mailjet.com]:587
RELAY_HOST=in-v3.mailjet.com
RELAY_PORT=587
RELAY_USER= # smtp username
RELAY_PASSWORD= # smtp password
```

### Relevant Stack Traces

```BASH
# BEGIN
May 16 18:55:13 mail postfix/smtp[2648]: Anonymous TLS connection established to in-v3.mailjet.com[104.199.96.85]:587: TLSv1.2 with cipher ADH-AES256-GCM-SHA384 (256/256 bits)
May 16 18:55:13 mail postfix/smtp[2648]: warning: SASL authentication failure: No worthy mechs found
May 16 18:55:13 mail postfix/smtp[2648]: B34B387A85506: to=<user@domain.tld>, relay=in-v3.mailjet.com[104.199.96.85]:587, delay=0.57, delays=0.07/0.06/0.45/0, dsn=4.7.0, status=deferred (SASL authentication failed; cannot authenticate to server in-v3.mailjet.com[104.199.96.85]: no mechanism available)
# END
```

**Comment by aendeavor** [comment]
Although I really much doubt this being the error's source, there seems to be a rather big discrepancy between your Docker version and your version of `docker-compose` (latest is 1.29).

If I had to put my fingers on it, either misconfiguration in some way (I'm not using a relay, others may be of better help here), or (from my POV more likely) an ARM package does not do what it is supposed to to.

I'd first try the `:edge` tag and see whether the problem disappears. If this is the case, it was likely a faulty arm package in `9.1.0`.

**Comment by ShinoYasx** [comment]
I tried with the edge version, it seems to be working. Regarding the docker-compose version, I'm using the version shipped by debian for Debian buster, so it may be a bit old but it shouldn't be a problem since the syntax is compatible.
So would  the debian sasl modules packages be the cause of the problem?

**Comment by aendeavor** [comment]
It could be that a package that was installed during built-time for `9.1.0` for ARM is faulty, yes. But TBH I'm not sure about that.

We could re-trigger the workflow for `9.1.0` to re-build the image, and since there is no version pinning on the packages that are to be installed themselves, we should get newer versions of them installed.

@wernerfred your take?

**Comment by wernerfred** [comment]
Sadly, one can only re-run gh actions if they had a failed job. Is it possible for you @ShinoYasx to stay on :edge (or rebuild with tag v9.1.0 yourself) until we release v10.0.0 (in roughly one week)?

**Comment by ShinoYasx** [comment]
No problem, I'll keep on `:edge` until a new release. If I find any bug I'll report it but for now it is working fine. Thank you !

**Comment by aendeavor** [comment]
> No problem, I'll keep on `:edge` until a new release. If I find any bug I'll report it but for now it is working fine. Thank you !

Alright. Version 10 is about the be release after some currents PRs are merged + plus one week of feature freezes. I'd estimate this time to be roughly 20 days.

**Comment by ShinoYasx** [comment]
Well... I got the same issue again, I don't know what I've done. I think I just updated the image and the error reappeared - the problem may be somewhere else.

I read on postfix documentation that the problem can be due to the relay auth method being not supported, but when I connect directly to the smtp server (in my case in-v3.mailjet.com) and I send EHLO, it responds `AUTH PLAIN LOGIN DIGEST-MD5 CRAM-MD5`, and the postfix configuration that this image provides permit plain login with `smtp_sasl_security_options = noanonymous`.

But the [postfix SASL documentation](http://www.postfix.org/SASL_README.html#client_sasl_policy) precises that a such error can occur when the `libplain.so` or `liblogin.so` are not installed in the `/usr/lib/sasl2` directory. Actually, `libsasl2-modules` provides these libraries but they are located in `/usr/lib/arm-linux-gnueabihf/sasl2/` since my host is arm. I tried to link `/usr/lib/arm-linux-gnueabihf/sasl2/*` to `/usr/lib/sasl2` but it doesn't change anything.

Any idea?

**Comment by aendeavor** [comment]
This is a very wild issue...

**Comment by ShinoYasx** [comment]
Okay, I think I found the problem and it's not related to docker-mailserver. It seems that the filesystem where docker in installed is somewhat broken and some files aren't recognized as regular file by the container. I think here postfix tries to load the sasl module but fails and throw the `No worthy mechs found` because it is not recognize as a regular file.

To confirm this, I tried to run another mailserver with the same configuration but with docker on the host filesystem instead of an XFS RAID and it worked fine. I noticed that the filesystem may be broken by reading the [docker OverlayFS documentation](https://docs.docker.com/storage/storagedriver/overlayfs-driver/) and when the `xfs_info` command thrown `/dev/sda is not a valid XFS filesystem`.

Sorry for taking your time with an issue for which I'm fully responsible. So I close this issue since it has been resolved.
